tests/unit/s2n_client_key_share_extension_pq_test.c

```diff
@@ -0,0 +1,1069 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include <stdint.h>
+
+#include "tls/s2n_config.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_tls13.h"
+#include "tls/extensions/s2n_client_key_share.h"
+#include "tls/extensions/s2n_key_share.h"
+#include "tls/s2n_security_policies.h"
+
+#include "testlib/s2n_testlib.h"
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_safety.h"
+#include "crypto/s2n_fips.h"
+
+#define HELLO_RETRY_MSG_NO 1
+
+#if defined(S2N_NO_PQ)
+
+int main() {
+    BEGIN_TEST();
+    END_TEST();
+    return 0;
+}
+
+#else
+
+static int s2n_generate_pq_hybrid_key_share_for_test(struct s2n_stuffer *out, struct s2n_kem_group_params *kem_group_params);
+
+int main() {
+    BEGIN_TEST();
+    /* PQ hybrid tests for s2n_client_key_share_extension */
```

Comment:
Everything from lines 48 to 380 was simply moved from `s2n_client_key_share_extension_test.c` with no material changes. (They are tests for `s2n_client_key_share_extension.send` which were added in https://github.com/awslabs/s2n/pull/2215 and https://github.com/awslabs/s2n/pull/2260.) New tests related to the functionality in this PR begin on line 381.